### DISCUSSION TITLE
Valetudo 2021.08.1

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1055] [Release Discussions]
<div align="center">
    <img src="https://github.com/Hypfer/Valetudo/blob/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.08.1</h2></p>
</div>

This release features the new React-based UI by @Jomik as well as some MQTT config changes

## New UI

The new UI has been merged and is now available as a usable preview.
There's still a lot to do, however that should be much easier now that it is merged.

Until everything has been ported over, the old UI will still be the default.

You can open the new one via the menu, which now also houses a shortcut to the Swagger UI as well:

![image](https://user-images.githubusercontent.com/974410/131392121-d28f2ae6-278d-49a3-a51e-2ca338107ec6.png)

![image](https://user-images.githubusercontent.com/974410/131392013-d984ab62-b3c4-4590-9081-f7f457b8ad11.png)

There's a drop-down where you can select a zone preset to clean in the new UI.

![image](https://user-images.githubusercontent.com/974410/131392187-40648196-50be-4c23-8cea-4ed1864decd5.png)

Futhermore, there's also a button, which lets you set the iteration count for segment cleanups.

![image](https://user-images.githubusercontent.com/974410/131392224-31302e3d-c8dc-43ce-823f-8c82732a6ffd.png)

This marks a huge step forward for the project, which has been due since years. Thanks again to @Jomik ❤

## MQTT config changes

This release removes all mqtt settings where I couldn't find a reasonable explanation for why they exist.
If you're missing something, please let me know in the comments and include an explanation _why_ it was needed.

## Dreame Z10 Pro Support

If you're looking for a Valetudo-supported and affordable Vacuum Robot featuring an Auto-Empty Dock, you can now buy
the Dreame Z10 Pro. I didn't know that I needed an auto-empty dock before I got one, but I definitely needed one.

Do note that it's supported on the feature level of the L10 Pro, meaning that you can't control the Dock yet.
That shouldn't matter though as it is on by default and doesn't strictly require any interactions.

### Features

* **ui:** Add link to swagger ui to menu ([513bd8c](https://github.com/Hypfer/Valetudo/commit/513bd8c2778ae2f26186ae48f69030bbfde59628))
* **ui:** Added html title to symbols in various map views ([#1026](https://github.com/Hypfer/Valetudo/issues/1026)) ([98d0f1e](https://github.com/Hypfer/Valetudo/commit/98d0f1e617cae48cbd44a171b5aa9178f0677a0b))
* **ui:** Display system host info ([54df5ca](https://github.com/Hypfer/Valetudo/commit/54df5ca4f6b53a280bad52f0a73f074e1d5122f3))
* **ui:** Support segment cleanup iteration count selection in map view ([f798ff5](https://github.com/Hypfer/Valetudo/commit/f798ff5d040e08a7b2f4d2606e61d8c5000957f5))
* **vendor.dreame:** Add Dreame Z10 Pro ([4127ccd](https://github.com/Hypfer/Valetudo/commit/4127ccdc413155c5f3d03a29b51fc7c2f10a1353))
* **vendor.dreame:** Log rootfs on startup if embedded ([cbde5a4](https://github.com/Hypfer/Valetudo/commit/cbde5a4873be1113b0dc5cfee0c1164cdc5bedf4))
* **vendor.roborock:** Log rootfs on startup if embedded ([8d09fd6](https://github.com/Hypfer/Valetudo/commit/8d09fd6dab8413b3b5e9aa89921014ef267b3f23))
* New React UI ([9acb2cd](https://github.com/Hypfer/Valetudo/commit/9acb2cd529a1b84dd7db333778dd523d8487c4e7))

### Bug Fixes

* **miio:** Add proper shutdown methods to dummycloud and retrywrapper ([f0676ef](https://github.com/Hypfer/Valetudo/commit/f0676efa6d81535069f156dab2611b63b7e4d27c))
* **miio:** Reduce loglevel for non-pending messages ([dbabe0b](https://github.com/Hypfer/Valetudo/commit/dbabe0b3a8e9540aff2d6f8f76dfdea2b68c767f))
* **mqtt:** Catch exceptions on connect reconfiguration ([0e3bebd](https://github.com/Hypfer/Valetudo/commit/0e3bebda27553ee7f7b3796a4b89dbd53eec1ccb))
* **mqtt:** Fix 100% cpu consumable fetch recursion ([ae163a9](https://github.com/Hypfer/Valetudo/commit/ae163a95e95b384e4e3e2804f9a4792c9e9793e9))
* **mqtt:** Remove refreshInterval setting ([83d2a45](https://github.com/Hypfer/Valetudo/commit/83d2a45eb0d21cc15cc58cdf1734833688de0fac))
* **networkadvertisement:** Hopefully fix valetudo crashing due to network issues ([4ace37e](https://github.com/Hypfer/Valetudo/commit/4ace37e0aaf88e5bb53dfbab5fbb4bf1d43c685e))
* **ui:** Fix minor issues discovered by deepscan ([ee8b58b](https://github.com/Hypfer/Valetudo/commit/ee8b58b646ce006d8c1d6a5cca2f27d6af32392c))
* **ui:** Only dark mode (for now?) ([db985f3](https://github.com/Hypfer/Valetudo/commit/db985f39fcfdf1bc9048d50a3e2d7d75bc42f3da))
* **vendor.dreame:** Add error code 68 mapping ([741c711](https://github.com/Hypfer/Valetudo/commit/741c7114d4b88760cd30ab7e50697816f968ed2a))
* **vendor.dreame:** Fix incorrect status mapping for powersave mode ([343c749](https://github.com/Hypfer/Valetudo/commit/343c749a107fdaa4f016965984065585c0872616))
* **vendor.dreame:** Ignore "event_occured" cloud messages ([4d3f4ac](https://github.com/Hypfer/Valetudo/commit/4d3f4acf8fd864d690850070328b5f9b115934d4))
* **vendor.roborock:** Handle more events ([972e4df](https://github.com/Hypfer/Valetudo/commit/972e4df9fc840c85324de7ef926763a521bd225d))
* **vendor.viomi:** Poll state every 30s to fix stale mqtt data ([fdb315a](https://github.com/Hypfer/Valetudo/commit/fdb315a33beb6690a68f34a131bd7484802ded7d))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.08.1'>Valetudo 2021.08.1</a>.</em>

**Comment by Baschdl578** [comment]
Just tried it out, the new UI looks and feels great, very responsive.
Two issues I ran into:
1. MQTT TLS:   
I have my MQTT server behind a proper domain with a proper CA-signed certificate. In the past, I just configured the domain and port, and enabled the now removed SSL/TLS option. No further config was necessary. With the SSL/TLS option removed, valetudo is not connecting to my MQTT server. I have not tried this yet, but I guess configuring the correct CA will now enable TLS? If that's true, it doesn't seem very straight-forward.
2. New UI:   
I know it is still WIP, just one general point of feedback. The robot's path and the segment names should be different colors. As it is right now, I can't read the room names (not that I need to, I know my own home), because they are white on a white path. It just looks bad, painting the path in black would fix this.

I really love the work you are doing, keep it up and thank you!

**Reply by Hypfer** [reply]
1)

Huh, interesting. I had no idea. I'm also not very happy with the google results for `mqtt tls`. 

I assumed that SSL/TLS requires a client certificate to authenticate which would've made the useTLS switch obsolete as it would be implied by a CA, a Cert and a Key being present. That's how it is now:
https://github.com/Hypfer/Valetudo/blob/db16144512f45c97b8512820609081a7696e62c5/backend/lib/mqtt/MqttController.js#L252

Thinking about that now, I wonder: are the CA, Cert and Key even related to each other or do we use the CA to verify the cert of the broker while we use the cert + key to authenticate ourself?
If so, I suspect that the cert/key options would be mutally exclusive to the user/password options?

Edit:

Looking at the previous PRs, that's exactly how it is. CA (#188) and Cert/Key (#549).
According to [this](http://www.steves-internet-guide.com/creating-and-using-client-certificates-with-mqtt-and-mosquitto/) Password and Certificate auth can also be used simultaneously. Whyever that is.

There are at least two lessons to be learned here

1. Don't merge PRs without proper documentation/comments
2. Contrary to the text of #188, it is possible to have TLS without specifying a cert.

Still, I think it was a good choice to remove it and see what happens as there's now the opportunity to do it right (e.g. by not having the MQTT config be a flat object. Also Docs! Docs!!!! DOCS!!!!!)


2)

Noted

**Reply by Baschdl578** [reply]
I have never used the cert/key method for authentication, I just use the certificate for encryption. But they might not be mutually exclusive in some special scenarios. Someone might have their broker behind a reverse proxy that uses cert/key auth and the broker itself still needs a user/pw. I'm not really sure tbh, just speculating here.

**Reply by ghost** [reply]
Thanks for the release! Cool to have zones back in the frontend again! :)

I saw that "topicPrefix" mqtt option was removed.
I really liked that option as i prefer to have a certain structure of my mqtt topics, grouped by either room or protocol etc. It quickly gets real messy if you have a lot of mqtt devices/services otherwise.
Any chance you can bring that back?

**Reply by matejzero** [reply]
Another thumbs up for Identifier (if it was removed) and topicPrefix. I have a custom mqtt structure to make sense of it and would like to keep it.

**Reply by Hypfer** [reply]
> grouped by either room or protocol 

Thanks for the explaination. 
I've readded the topicPrefix and also documented this reason in the schema:
https://github.com/Hypfer/Valetudo/commit/f00edb32767a01ed71455fab13b57c205cf4dae2#diff-835981aa52b3670e3a248338793080220acf0a6202f6c4fd1c1d2a5a7448ffc7R192

**Comment by PhilipWhiteside** [comment]
Thanks for the release, the new GUI looks great.

Was there any further changes to the MQTT?  I was using `MapSegmentationCapability` with a string number payload, this worked fine before, even earlier today when the automation ran from HomeAssistant. Once I updated the payload had to be JSON. I don't have any concerns using JSON, maybe this can help others who get a bit stuck also, save them some troubleshooting time. 

The documentation does say to use JSON. Where I had `payload: "18"` before I now have `payload: '{"segment_ids": ["18"], "iterations": 1, "customOrder": true}'` as per the docs https://valetudo.cloud/pages/integrations/home-assistant-integration

```yaml
  - name: Clean kitchen
    service: mqtt.publish
    service_data:
      topic: valetudo/vacuum-01/MapSegmentationCapability/clean/set
      payload: '{"segment_ids": ["18"], "iterations": 1, "customOrder": true}'
    icon: mdi:silverware-fork-knife
```
